packages/backend/src/services/RenameService/rename.ts

```diff
@@ -0,0 +1,666 @@
+    FilterGroupItem,
+    Filters,
+    isAndFilterGroup,
+    isCustomBinDimension,
+    isFilterRule,
+    isOrFilterGroup,
+    MetricQuery,
+    NameChanges,
+    RenameType,
+    SavedChartDAO,
+    SchedulerAndTargets,
+} from '@lightdash/common';
+
+/* There are different methods to replace model names
+ replaceId: Replace only the start of a string, with the new model prefix
+ replaceReference: Replace the reference string part (eg: ${payment.) with the new model name, needed for SQL strings and more complex references
+ replaceString: Do a full string replacement with the new model name, used only if others don't apply, used in custom viz (vega-lite)
+ and some other functions to deal with undefined, lists and objects
+*/
+export const createRenameFactory = (
+    from: string,
+    to: string,
+    fromReference: string,
+    toReference: string,
+    isPrefix: boolean,
+) => {
+    let replaceId: (str: string) => string;
+    let replaceReference: (str: string) => string;
+    let replaceString: (str: string) => string;
+    if (isPrefix) {
+        replaceId = (str: string) =>
+            str.replace(new RegExp(`^${from}_`, 'g'), `${to}_`); // table prefix (eg: payment_)
+        replaceReference = (str: string) =>
+            str.replace(
+                new RegExp(`\\$\\{${fromReference}\\.`, 'g'),
+                `\${${toReference}.`,
+            ); // SQL normally uses "." on references
+        replaceString = (str: string) =>
+            str.replace(new RegExp(`${from}_`, 'g'), `${to}_`);
+    } else {
+        replaceId = (str: string) =>
+            str.replace(new RegExp(`^${from}$`, 'g'), `${to}`); // entire id (eg: payment_customer_id)
+        replaceReference = (str: string) =>
+            str.replace(
+                new RegExp(`\\$\\{${fromReference}\\}`, 'g'),
+                `\${${toReference}}`,
+            ); // Full reference
+        replaceString = (str: string) =>
+            str.replace(new RegExp(`${from}`, 'g'), `${to}`);
+    }
+
+    return {
+        replaceId,
+        replaceString,
+        replaceReference,
+        replaceOptionalId: (str: string | undefined) =>
+            str ? replaceId(str) : undefined,
+        replaceKeys: <T>(obj: Record<string, T>) => {
+            const keyValues = Object.keys(obj).map((key) => {
+                const newKey = replaceId(key);
+                return { [newKey]: obj[key] };
+            });
+            return Object.assign({}, ...keyValues);
+        },
+        replaceFull: (str: string) => str.replaceAll(from, to), // Full name replace, used in tables
+        replaceList: (list: string[]) => list.map((item) => replaceId(item)),
+        replaceOptionalList: (list: string[] | undefined) =>
+            list ? list.map((item) => replaceId(item)) : undefined,
+    };
+};
+
+/*
+We are renaming fields in the object 
+This is only for validating that the we are replacing all the fields needed
+*/
+export const validateRename = (
```

Comment:
Useful method for checking differences, I'm planning to add some monitoring to let us know if we miss something. 

Renaming using JSON is sometimes is innacurate, and could lead to false positives

Reply:
Depending on the impact on production, I might remove it later